rclcpp_components/src/component_manager_isolated.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp_components/component_manager_isolated.hpp"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+namespace rclcpp_components
+{
+
+ComponentManagerIsolated::~ComponentManagerIsolated()
+{
+  if (node_wrappers_.size()) {
+    for (auto & prom : dedicated_promises_) {
+      prom.second.set_value();
+    }
+    for (auto & exec : dedicated_executors_) {
+      exec.second->cancel();
+    }
+    for (auto & t : dedicated_threads_) {
+      t.second.join();
+    }
+    node_wrappers_.clear();
+  }
+}
+
+void
+ComponentManagerIsolated::on_load_node(
```

Comment:
Some of the code here is repeated from the base class and could be shared with it.
Same in on_unload_node().

Anyways the approach seems reasonable!

Reply:
Feel free to ignore this comment